osquery/tables/system/windows/bitlocker_info.cpp

```diff
@@ -26,24 +26,53 @@ QueryData genBitlockerInfo(QueryContext& context) {
       "SELECT * FROM Win32_EncryptableVolume",
       (BSTR)L"ROOT\\CIMV2\\Security\\MicrosoftVolumeEncryption");
   std::vector<WmiResultItem>& wmiResults = wmiSystemReq.results();
-  if (!wmiResults.empty()) {
-    for (const auto& data : wmiResults) {
-      long status = 0;
-      data.GetString("DeviceID", r["device_id"]);
-      data.GetString("DriveLetter", r["drive_letter"]);
-      data.GetString("PersistentVolumeID", r["persistent_volume_id"]);
-      data.GetLong("ConversionStatus", status);
-      r["conversion_status"] = INTEGER(status);
-      data.GetLong("ProtectionStatus", status);
-      r["protection_status"] = INTEGER(status);
-      results.push_back(r);
+  if (wmiResults.empty()) {
+    LOG(WARNING) << "Error retreiving information from WMI.";
+    return results;
+  }
+  for (const auto& data : wmiResults) {
+    long status = 0;
+    long emethod;
+    std::string emethod_str;
+    data.GetString("DeviceID", r["device_id"]);
+    data.GetString("DriveLetter", r["drive_letter"]);
+    data.GetString("PersistentVolumeID", r["persistent_volume_id"]);
+    data.GetLong("ConversionStatus", status);
+    r["conversion_status"] = INTEGER(status);
+    data.GetLong("ProtectionStatus", status);
+    r["protection_status"] = INTEGER(status);
+    data.GetLong("EncryptionMethod", emethod);
+    switch (emethod) {
```

Comment:
Nit: Could you bring emethod_str down to here so it's closer to usage, or set r["encryption_method"] within the cases?

Reply:
You mean like move the declaration of it between current lines 44 and 45? Yeah I can do that